readthedocs/projects/tasks.py

```diff
@@ -743,15 +744,23 @@ def fileify(version_pk, commit):
                 if fnmatch.fnmatch(filename, '*.html'):
                     dirpath = os.path.join(root.replace(path, '').lstrip('/'),
                                            filename.lstrip('/'))
-                    obj, created = ImportedFile.objects.get_or_create(
-                        project=project,
-                        version=version,
-                        path=dirpath,
-                        name=filename,
-                        commit=commit,
-                    )
-                    if not created:
-                        obj.save()
+                    full_path = os.path.join(root, filename)
+                    md5 = hashlib.md5(open(full_path, 'rb').read()).hexdigest()
+                    try:
+                        obj, created = ImportedFile.objects.get_or_create(
+                            project=project,
+                            version=version,
+                            path=dirpath,
+                            name=filename,
+                        )
+                    except Exception:
+                        log.exception('Error creating ImportedFile')
+                        continue
+                    if obj.md5 != md5:
+                        obj.md5 = md5
+                    if obj.commit != commit:
```

Comment:
I think the two `if`s are not required here unless assigning a value has any sideeffect here.